fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java

```diff
@@ -1317,4 +1326,112 @@ public TSetConfigResponse setConfig(TSetConfigRequest request) throws TException
+            if (db != null) {
+                db.readLock();
+                try {
+                    List<Table> allTables = db.getTables();
+                    allTables.forEach(table -> {
+                        TTableMetaInfo tableMetaInfo = new TTableMetaInfo();
+                        tableMetaInfo.setTable_schema(dbName);
+                        tableMetaInfo.setTable_name(table.getName());
+                        
+                        if (table.isOlapOrLakeTable() || 
+                                table.getType() == TableType.OLAP_EXTERNAL ||
+                                table.getType() == TableType.MATERIALIZED_VIEW) {
+                            // OLAP (done)
+                            // OLAP_EXTERNAL (done)
+                            // MATERIALIZED_VIEW (done)
+                            // LAKE (done)
+                            tableMetaInfo = genNormalTableMetaInfo(table, tableMetaInfo);
+                        } else {
+                            // SCHEMA (use default)
+                            // INLINE_VIEW (use default)
+                            // VIEW (use default)
+                            // BROKER (use default)                           
+                            tableMetaInfo = genDefaultMetaInfo(tableMetaInfo);
+                        }
+                        // TODO(cjs): other table type (HIVE, MYSQL, ICEBERG, HUDI, JDBC, ELASTICSEARCH)
+                        tList.add(tableMetaInfo);
+                    });
+                } finally {
+                    db.readUnlock();
+                }                
+            }            
+        });
+        resp.tables_meta_infos = tList;
+        return resp;
+    }
+
+    static final ImmutableList<String> SHOW_PROPERTIES_NAME = new ImmutableList.Builder<String>()
+            .add(PropertyAnalyzer.PROPERTIES_STORAGE_TYPE)
+            .add(PropertyAnalyzer.PROPERTIES_STORAGE_MEDIUM)
+            .add(PropertyAnalyzer.PROPERTIES_STORAGE_COLDOWN_TIME)
+            .build();
+
+    private TTableMetaInfo genNormalTableMetaInfo(Table table, TTableMetaInfo tableMetaInfo) {
+        OlapTable olapTable = (OlapTable) table;
+        // Distribution info
+        DistributionInfo distributionInfo = olapTable.getDefaultDistributionInfo();
+        String distributeKey = distributionInfo.getDistributionKey();
+        String distributeNum = String.valueOf(distributionInfo.getBucketNum());
+        // Partition info
+        PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+        StringBuilder partitionKeySb = new StringBuilder();
+        int idx = 0;
+        try {
+            for (Column column : partitionInfo.getPartitionColumns()) {
+                if (idx != 0) {
+                    partitionKeySb.append(", ");
+                }
+                partitionKeySb.append("`").append(column.getName()).append("`");
+                idx++;
+            }
+        } catch (NotImplementedException e) {
```

Comment:
NotImplementedException should be regarded as a bug, I suggest you check partitionInfo.getType() instead.

Reply:
done